src/librustdoc/clean/utils.rs

```diff
@@ -453,3 +454,18 @@ crate fn find_nearest_parent_module(tcx: TyCtxt<'_>, def_id: DefId) -> Option<De
         None
     }
 }
+
+/// Checks that one attribute is `doc`. For example:
+///
+/// ```text
+/// #[doc(spotlight)]
+/// ```
+///
+/// This function has to exists because it runs on `hir::Attributes` whereas the other runs on
+/// `clean::Attributes`.
+crate fn has_doc_flag(attrs: Attributes<'_>, flag: Symbol) -> bool {
+    attrs.iter().any(|attr| {
+        attr.has_name(sym::doc)
+            && attr.meta_item_list().map_or(false, |l| list_contains_name(&l, flag))
```

Comment:
`list_contains_name` is a very generic function name. Maybe use a qualified path instead?

```suggestion
            && attr.meta_item_list().map_or(false, |l| rustc_attr::list_contains_name(&l, flag))
```